code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java

```diff
@@ -178,7 +181,134 @@ private void dispatchTrackRequest(final Map<String, Object> data, final Event pa
 		getApi().dispatch(event);
 	}
 
-	private boolean isNullOrEmpty(final Map map) {
-		return map == null || map.isEmpty();
+	/**
+	 * Formats track event data to the required Analytics Edge translator format under the `data.__adobe.analytics` object.
+	 *
+	 * The following is the mapping logic:
+	 * - The "action" field is mapped to "data.__adobe.analytics.linkName", and "data.__adobe.analytics.linkType" is set to "other".
+	 * - The "state" field is mapped to "data.__adobe.analytics.pageName".
+	 * - Any "contextData" keys that start with the "&&" prefix are mapped to "data.__adobe.analytics" with the prefix removed.
+	 * - Any "contextData" keys without the "&&" prefix are mapped to "data.__adobe.analytics.contextData".
+	 * - Any additional fields are passed through and left directly under the "data" object.
+	 *
+	 * As an example, the following track event data:
+	 * ```
+	 * {
+	 *    "action": "action name",
+	 *    "contextData": {
+	 *       "&&c1": "propValue1",
+	 *       "key1": "value1"
+	 *    },
+	 *    "key2": "value2"
+	 * }
+	 * ```
+	 * Is mapped to:
+	 * ```
+	 * {
+	 *   "data": {
+	 *     "__adobe": {
+	 *       "analytics": {
+	 *         "linkName": "action name",
+	 *         "linkType": "other",
+	 *         "c1": "propValue1",
+	 *         "contextData": {
+	 *           "key1": "value1"
+	 *         }
+	 *       }
+	 *     },
+	 *     "key2": "value2"
+	 *   }
+	 * }
+	 * ```
+	 *
+	 * @param data track event data
+	 * @return data formatted for the Analytics Edge translator.
+	 */
+	private Map<String, Object> formatData(Map<String, Object> data) {
+		// Create a mutable copy of data
+		Map<String, Object> mutableData = deepCopy(data);
+		// __adobe.analytics data container
+		Map<String, Object> analyticsData = new HashMap<>();
+
+		// Extract contextData
+		final Map<String, Object> contextData = DataReader.optTypedMap(
+			Object.class,
+			mutableData,
+			EdgeBridgeConstants.AnalyticsKeys.CONTEXT_DATA,
+			null
+		);
+		mutableData.remove(EdgeBridgeConstants.AnalyticsKeys.CONTEXT_DATA);
```

Comment:
npe when deepCopy returned null. we should handle the case when the clone did not succeed and fast fail

Reply:
In this case I see it as an exception case, it means data is incorrectly formatted and we cannot convert it to the new format. We can log warning and don't send any request out.